aspnetcore/migration/webapi.md

```diff
@@ -4,47 +4,164 @@ author: ardalis
+
+The *ProductsApp* project includes one controller. The controller inherits from `ApiController` and contains two actions:
+
+[!code-csharp[](webapi/sample/3.x/ProductsApp/Controllers/ProductsController.cs?highlight=28,33)]
+
+The `Product` model used by `ProductsController` is a basic class:
+
+[!code-csharp[](webapi/sample/3.x/ProductsApp/Models/Product.cs)]
+
+The following sections demonstrate migration of the Web API project to ASP.NET Core MVC.
+
+## Create the destination project
+
+Create a new blank solution in Visual Studio and add the ASP.NET 4.x Web API project to migrate:
+
+1. From the **File** menu, select **New** > **Project**.
+1. Select the **Blank Solution** template and select **Next**.
+1. Name the solution *WebAPIMigration*. Select **Create**.
+1. Add the existing *ProductsApp* project to the solution.
+
+Add a new API project to migrate to:
+
+1. Add a new **ASP.NET Core Web Application** project to the solution.
+1. In the **Configure your new project** dialog, Name the project *ProductsCore*, and select **Create**.
+1. In the **Create a new ASP.NET Core Web Application** dialog, confirm that **.NET Core** and **ASP.NET Core 3.1** are selected. Select the **API** project template, and select **Create**.
+1. Remove the *WeatherForecast.cs* and *Controllers/WeatherForecastController.cs* example files from the new *ProductsCore* project.
+
+The solution now contains two projects. The following sections explain migrating the *ProductsApp* project's contents to the *ProductsCore* project.
+
+## Migrate configuration
+
+ASP.NET Core doesn't use the *App_Start* folder or the *Global.asax* file, and the *web.config* file is added at publish time. *Startup.cs* is the replacement for *Global.asax* and is located in the project root. The `Startup` class handles all app startup tasks. For more information, see <xref:fundamentals/startup>.
+
+All ASP.NET Core project templates include attribute routing in the generated code. The following `UseRouting` and `UseEndpoints` calls register route matching and endpoint execution in the [middleware](xref:fundamentals/middleware/index) pipeline. They replace the *ProductsApp* project's *App_Start/WebApiConfig.cs* file:
+
+[!code-csharp[](webapi/sample/3.x/ProductsCore/Startup.cs?name=snippet_Configure&highlight=10,14)]
+
+## Migrate models and controllers
+
+Copy over the *ProductApp* project's controller and the model it uses. Follow these steps:
+
+1. Copy *Controllers/ProductsController.cs* from the original project to the new one.
+1. Copy the entire *Models* folder from the original project to the new one.
+1. Change the copied files' namespaces to match the new project name (*ProductsCore*). Adjust the `using ProductsApp.Models;` statement in *ProductsController.cs* too.
+
+At this point, building the app results in a number of compilation errors. The errors occur because the following components don't exist in ASP.NET Core:
```

Comment:
Why are you telling them this in the middle of the migration. Just keep going.

Reply:
It was original to the instruction that was already reviewed, as many of these were so I left it in.  I actually found it helpful since the first thing I did out of habit was compile to see what shape they were in and ran into the errors, then saw that was expected.  It is likely most folks will not do that however and assume they just need to keep going through steps and not be concerned.  So I made the changes.